greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java

```diff
@@ -29,22 +32,30 @@
 @Component
 public class NameCacheUtil {
 
-    private final Cache springCache;
+    private static final String KEY = "key";
+
+    private final int ttlSeconds;
     private final BasicCache<String, String> infispanCache;
 
-    public NameCacheUtil(CacheManager cacheManager) {
-        springCache = cacheManager.getCache(NAME_CACHE_ID);
+    public NameCacheUtil(@Value("${cache.ttl}") int ttlSeconds, CacheManager cacheManager) {
+        this.ttlSeconds = ttlSeconds;
+        final Cache springCache = cacheManager.getCache(NAME_CACHE_ID);
         infispanCache = (BasicCache<String, String>) springCache.getNativeCache();
     }
 
-    /**
-     * Clears all entries found in nameCache
-     */
     public void clear() {
-        springCache.clear();
+        infispanCache.clear();
     }
 
     public boolean isEmpty() {
         return infispanCache.isEmpty();
     }
+
+    public String get() {
+        return infispanCache.get(KEY);
+    }
+
+    public void put(String value) {
+        infispanCache.put(KEY, value, ttlSeconds, TimeUnit.MILLISECONDS);
```

Comment:
If `ttlSeconds` is in seconds, then `TimeUnit.MILLISECONDS` is wrong :-)

Reply:
And I see below that `cache.ttl` is actually in millis -- so it's the variable name `ttlSeconds` what is wrong.

Reply:
Yup, I changed it after fixing the tests, but obviously didn't fix it everywhere :)